src/ui/public/visualize/visualize.js

```diff
@@ -44,6 +44,19 @@ define(function (require) {
         var getVisEl = getter('.visualize-chart');
         var getVisContainer = getter('.vis-container');
 
+        // Show no results message when hasZeroHits is true and it requires search
+        $scope.showNoResultsMessage = function () {
+          var requiresSearch = _.get($scope, 'vis.type.requiresSearch');
```

Comment:
my bad, I didn't notice these were both already booleans.  it looks like the has check works fine but probably wasn't needed